packages/data-broker/src/DataBroker.ts

```diff
@@ -144,33 +153,70 @@ export class DataBroker {
     return syncGroups.map(sg => ({ ...sg, type: this.getDataSourceTypes().SYNC_GROUP }));
   };
 
-  private checkDataElementPermissions = async (dataElements: DataSource[]) => {
-    const userPermissions = await this.getUserPermissions();
-    if (userPermissions.includes(BES_ADMIN_PERMISSION_GROUP)) {
+  private getCountryCodes = async (organisationUnitCodes: string[]) => {
+    const orgUnits = await this.models.entity.find({ code: organisationUnitCodes });
+    const orgUnitCountryCodes = orgUnits.map(orgUnit => orgUnit.country_code).filter(isNotNullish);
+    const countryCodes = [...new Set(orgUnitCountryCodes)];
+    return countryCodes;
+  };
+
+  private checkDataElementPermissions = async (
+    dataElements: DataSource[],
+    countryCodes: string[] | null,
+  ) => {
+    const allUserPermissions = this.getUserPermissions();
+    if (allUserPermissions.includes(BES_ADMIN_PERMISSION_GROUP)) {
       return true;
     }
-    const missingPermissions = [];
-    for (const element of dataElements as DataElement[]) {
-      if (
-        element.permission_groups.length <= 0 ||
-        element.permission_groups.some(code => userPermissions.includes(code))
-      ) {
-        continue;
+
+    const getDataElementsWithMissingPermissions = (permissions: string[]) =>
+      (dataElements as DataElement[])
+        .filter(element => element.permission_groups.length > 0)
+        .filter(element => !element.permission_groups.some(group => permissions.includes(group)))
+        .map(element => element.code);
+
+    if (countryCodes === null) {
```

Comment:
I'd love to drop support for no country codes if possible, I can't wrap my head around it. In fact I don't know if it's ever done?

Reply:
Ah just saw your other comment, all good.